index.js

```diff
@@ -20,7 +20,12 @@ function arrayEquals(as, bs) {
 }
 
 function addToLookup(zigbeeModel, definition) {
-    zigbeeModel = zigbeeModel ? zigbeeModel.toLowerCase() : null;
+    if (zigbeeModel) {
+        zigbeeModel = String(zigbeeModel);
```

Comment:
I don't understand why this is needed, in the example from https://github.com/Koenkk/zigbee-herdsman-converters/issues/2720#issuecomment-869010706 `zigbeeModel` is already a string.

Reply:
Maybe.
But only this helped in my case.

I can try to reproduce the error, but only on Monday.

Reply:
I think this is my mistake in studying the device.
I have chosen the wrong encoding of some fields from the genBasic cluster. Because of what the functions did not work correctly.
Only after making edits to the code and subsequent firmware of the device, z2m started working correctly.

`{BASIC, {ATTRID_BASIC_DATE_CODE, ZCL_DATATYPE_CHAR_STR, R, (void *)zclGenericApp_DateCode}}`

`ZCL_UINT8` instead of `ZCL_DATATYPE_ENUM8`

Correct `ZCL_DATATYPE_ENUM8`

With the edits I made to the converter file index, I was able to spot this error.

Accordingly, whether to make these changes to the master branch or not is up to you.

Thank you for your responsiveness!

Reply:
Good, since the `zigbeeModel` should always be a string this is not needed, I will close it.